main.go

```diff
@@ -16,33 +16,269 @@ limitations under the License.
 
 package main
 
-import "github.com/googleinterns/k8s-topology-simulator/modeling"
+import (
+	"encoding/csv"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/googleinterns/k8s-topology-simulator/modeling"
+)
 
 func main() {
-	zoneA := modeling.Zone{Nodes: 30, Endpoints: 60, Name: "a"}
-	zoneB := modeling.Zone{Nodes: 35, Endpoints: 70, Name: "b"}
-	zoneC := modeling.Zone{Nodes: 50, Endpoints: 80, Name: "c"}
-	zones := []modeling.Zone{zoneA, zoneB, zoneC}
+	// algorithm name, default shared global
+	algPtr := flag.String("alg", "SharedGlobalAlgorithm", "routing algorithm")
+	// input file
+	inputPtr := flag.String("input", "test.csv", "info of each zone")
```

Comment:
Maybe create an `examples` folder that includes sample input and output?
```suggestion
	inputPtr := flag.String("input", "examples/input.csv", "Inputs to use for this algorithm")
```

Reply:
fixed